lib/docker/artifact_image.js

```diff
@@ -62,24 +78,38 @@ export default class ArtifactImage {
   }
 
   async _downloadArtifact() {
+    let tarballPath;
     let start = Date.now();
     let downloadDir = path.join(this.runtime.dockerVolume, 'tmp-docker-images', slugid.nice());
     await makeDir(downloadDir);
 
-    let originalTarball = path.join(downloadDir, 'image.tar');
+    let downloadedFile = path.join(downloadDir, path.basename(this.artifactPath));
 
     try {
       await downloadArtifact(
           this.task.queue,
           this.stream,
           this.taskId,
           this.artifactPath,
-          originalTarball,
+          downloadedFile,
           this.runtime.dockerConfig
       );
       this.runtime.monitor.measure('task.taskImage.downloadTime', Date.now() - start);
 
-      await this.renameAndLoad(this.imageName, originalTarball);
+      switch (path.extname(downloadedFile)) {
+        case '.lz4':
+          this.stream.write(fmtLog('Decompressing downloaded image'));
+          tarballPath = await decompressLz4File(downloadedFile);
+          fs.unlink(downloadedFile);
```

Comment:
In case of `decompressLz4File` fails, `downloadFile` but get stuck in the fs. I suggest using `try`/`finally` to guarantee the file is unliked.